Content.Shared/Item/ItemToggle/SharedItemToggleSystem.cs

```diff
@@ -0,0 +1,313 @@
+    {
+        if (args.Handled)
+            return;
+
+        args.Handled = true;
+
+        if (TryComp<WieldableComponent>(uid, out var wieldableComp))
+            return;
+
+        Toggle(uid, args.User, itemToggle);
+    }
+
+    public void Toggle(EntityUid uid, EntityUid? user = null, ItemToggleComponent? itemToggle = null)
+    {
+        if (!Resolve(uid, ref itemToggle))
+            return;
+
+        if (itemToggle.Activated)
+        {
+            TryDeactivate(uid, user, itemToggle: itemToggle);
+        }
+        else
+        {
+            TryActivate(uid, user, itemToggle: itemToggle);
+        }
+    }
+
+    public void ForceToggle(EntityUid uid, ItemToggleComponent itemToggle, ref ItemToggleForceToggleEvent args)
+    {
+        Toggle(uid, args.User, itemToggle);
+    }
+
+    public bool TryActivate(EntityUid uid, EntityUid? user = null, ItemToggleComponent? itemToggle = null)
+    {
+        if (!Resolve(uid, ref itemToggle))
+            return false;
+
+        if (itemToggle.Activated)
+            return true;
+
+        // The client cannot predict if the attempt to turn on fails or not since the battery and fuel systems are server side (for now). Potential future TODO
+        if (_netManager.IsServer)
+        {
+            var attempt = new ItemToggleActivateAttemptEvent(user);
+            RaiseLocalEvent(uid, ref attempt);
+
+            if (attempt.Cancelled)
+            {
+                //Play the failure to activate noise if there is any.
+                _audio.PlayPvs(itemToggle.SoundFailToActivate, uid);
+                return false;
+            }
+
+            // At this point the server knows that the activation went through successfully, so we play the sounds and make the changes.
+            _audio.PlayPvs(itemToggle.SoundActivate, uid);
+
+            // Starts the active sound (like humming).
+            if (TryComp(uid, out ItemToggleActiveSoundComponent? activeSound))
+            {
+                if (activeSound.ActiveSound != null && activeSound.PlayingStream == null)
+                {
+                    activeSound.PlayingStream = _audio.PlayPvs(activeSound.ActiveSound, uid, AudioParams.Default.WithLoop(true)).Value.Entity;
+                }
+            }
+
+            Activate(uid, itemToggle);
+            var ev = new ItemToggleActivatedEvent();
+            RaiseLocalEvent(uid, ref ev);
+        }
+        return true;
+    }
+
+    public bool TryDeactivate(EntityUid uid, EntityUid? user = null, ItemToggleComponent? itemToggle = null)
+    {
+        if (!Resolve(uid, ref itemToggle))
+            return false;
+
+        if (!itemToggle.Activated)
+            return true;
+
+        //Since there is currently no system that cancels a deactivation, it's all predicted.
+        var attempt = new ItemToggleDeactivateAttemptEvent(user);
+        RaiseLocalEvent(uid, ref attempt);
+
+        if (attempt.Cancelled && uid.Id == GetNetEntity(uid).Id)
```

Comment:
Uhh is this supposed to be here? It's not doing anything.

Reply:
It's a bit older - I used the comparison to make sure the entity performing the deactivation was the server one (hence having the netEntity uid). Removed it.